### PR TITLE
i652b8

### PR DESCRIPTION
**Pull request by s1mark**

### CODE REVIEW

File: Matuch Richard - I652B8/beadando/project/main.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module "module-i652b8" {
```
Comment (s1mark) [inline Matuch Richard - I652B8/beadando/project/main.tf:1]: looks nice

File: Matuch Richard - I652B8/beadando/module/vm/main.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  zone         = var.zone
+
+  boot_disk {
+    resource.boot_disk
+  }
+
+  network_interface {
+    network = "default"
+  }
+
```
Comment (s1mark) [inline Matuch Richard - I652B8/beadando/module/vm/main.tf:13]: missing the file creation

**Comment by R1chard2001** [comment]
Did "terraform apply", then "terraform destroy"

**Comment by R1chard2001** [comment]
Everything works as expected

File: Matuch Richard - I652B8/beadando/module/vm/provider.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+provider "google" {
```
[the remark this answers is not in the document]
Comment (s1mark) [reply Matuch Richard - I652B8/beadando/module/vm/provider.tf:1]: You are still having the double provider declarations

**Comment by R1chard2001** [comment]
Apply result:
```
data.google_project.my_project: Reading...
data.google_project.my_project: Read complete after 1s [id=projects/ekke-cloud]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # module.module-i652b8.google_compute_disk.attached_disk will be created
  + resource "google_compute_disk" "attached_disk" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "disk-i652b8"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # module.module-i652b8.google_compute_instance.compute_vm will be created
  + resource "google_compute_instance" "compute_vm" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done > ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "ekke-cloud-i652b8"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + attached_disk {
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)
        }

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = "debian-cloud/debian-11"
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # module.module-i652b8.google_service_account.service_account will be created
  + resource "google_service_account" "service_account" {
      + account_id = "sa-i652b8"
      + disabled   = false
      + email      = (known after apply)
      + id         = (known after apply)
      + member     = (known after apply)
      + name       = (known after apply)
      + project    = (known after apply)
      + unique_id  = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + compute_vm_external_ip = (known after apply)
  + compute_vm_id          = (known after apply)
  + project_id             = "ekke-cloud"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.module-i652b8.google_service_account.service_account: Creating...
module.module-i652b8.google_compute_disk.attached_disk: Creating...
module.module-i652b8.google_service_account.service_account: Creation complete after 2s [id=projects/ekke-cloud/serviceAccounts/sa-i652b8@ekke-cloud.iam.gserviceaccount.com]
module.module-i652b8.google_compute_disk.attached_disk: Still creating... [10s elapsed]
module.module-i652b8.google_compute_disk.attached_disk: Creation complete after 12s [id=projects/ekke-cloud/zones/europe-central2-a/disks/disk-i652b8]
module.module-i652b8.google_compute_instance.compute_vm: Creating...
module.module-i652b8.google_compute_instance.compute_vm: Still creating... [10s elapsed]
module.module-i652b8.google_compute_instance.compute_vm: Creation complete after 13s [id=projects/ekke-cloud/zones/europe-central2-a/instances/ekke-cloud-i652b8]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

compute_vm_external_ip = "10.186.0.55"
compute_vm_id = "projects/ekke-cloud/zones/europe-central2-a/instances/ekke-cloud-i652b8"
project_id = "ekke-cloud"
```

**Pull request merged**